### PR TITLE
Exosuit Proto-kinetic Accelerator

### PR DESCRIPTION
**Pull request by InvalidArgument3**
## About The Pull Request
Adds new equipment for mechas: the Exosuit Proto-kinetic Accelerator, unlocked with the mining tech research. Compared to the standard one, it is able to hit 5 tiles away with a low cooldown.

However, unlike the handheld version, it draws a considerable amount of power from the battery, pretty much requiring you to bring generator equipment or upgraded parts.

The projectile is identical to the standard handheld version otherwise and will do little damage in pressurized environments. 
Sprite:
dreammaker_2021-07-22_08-09-12

Video demo:
https://user-images.githubusercontent.com/51190031/126648230-3e0d10d4-4527-4267-bc94-378ec5ecaed7.mp4

## Why It's Good For The Game
According to the MODsuits document https://hackmd.io/@tgstation/SypLCs4jU, mechs are planned to be moved to a lavaland-only role, and I think this new equipment helps them accommodate.

## Changelog
:cl:
add: Exosuit-mounted kinetic accelerator equipment. Unlocked through Mining Technology.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
[#42737 (comment)](https://github.com/tgstation/tgstation/pull/42737#issuecomment-462999467)

**Comment by AndroidMK2** [comment]
It will make the mining mechs actually useful now, right now they are pretty much early round advanced plasma cutters for miners and once they get those no one will use the mechs for the rest of the round. This would add a extra utility to the mining mechs since the combat mechs like durand or gygax are literal dog poop when it comes to megafauna and even regular fauna killing.

**Comment by InvalidArgument3** [comment]
> It will make the mining mechs actually useful now, right now they are pretty much early round advanced plasma cutters for miners and once they get those no one will use the mechs for the rest of the round. This would add a extra utility to the mining mechs since the combat mechs like durand or gygax are literal dog poop when it comes to megafauna and even regular fauna killing.

I wouldn't get too excited, but it does bring them from "completely useless" to "mediocre" at dealing with basic lavaland mobs. As it can only fire in a ~130 degree(?) cone the pilot will have to make good use of strafing. 

Interestingly it's actually quite bad on the clarke because of this reason.

**Review by Twaticus — APPROVED** [review]

File: code/modules/vehicles/mecha/equipment/weapons/weapons.dm [hunk 1]
```diff
@@ -134,6 +134,25 @@
...
+	icon_state = "mecha_kineticgun"
+	energy_drain = 30
+	projectile = /obj/projectile/kinetic/mech
+	fire_sound = 'sound/weapons/kenetic_accel.ogg'
+	harmful = TRUE
+
+//only attachable to non-combat mechs
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/mecha_kineticgun/can_attach(obj/vehicle/sealed/mecha/M)
+	if(..()) //combat mech
+		return TRUE
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/equipment/weapons/weapons.dm:151]: This comment is incorrect as you copy pasted it from plasmacutters

Comment (InvalidArgument3) [reply code/modules/vehicles/mecha/equipment/weapons/weapons.dm]: I forgot that cutters were universal, thanks. Maybe this equipment should be too

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 2]
```diff
@@ -230,6 +230,8 @@
 	var/obj/effect/temp_visual/kinetic_blast/K = new /obj/effect/temp_visual/kinetic_blast(target_turf)
 	K.color = color
 
+/obj/projectile/kinetic/mech/ //mecha_kineticgun version of the projectile
```
Comment (Rohesie) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:233]:
```suggestion
//mecha_kineticgun version of the projectile
/obj/projectile/kinetic/mech
```
Avoid a `/` at the end of the path, and careful about comments after it.
we've had some funky types resulting from this pattern.

File: code/modules/vehicles/mecha/equipment/weapons/weapons.dm [hunk 3]
```diff
@@ -134,6 +134,25 @@
...
+	fire_sound = 'sound/weapons/kenetic_accel.ogg'
+	harmful = TRUE
+
+//attachable to all mechas, like the plasma cutter
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/mecha_kineticgun/can_attach(obj/vehicle/sealed/mecha/M)
+	if(..()) //combat mech
+		return TRUE
+	else if(LAZYLEN(M.equipment) < M.max_equip)
+		return TRUE
+	return FALSE
```
Comment (Rohesie) [inline code/modules/vehicles/mecha/equipment/weapons/weapons.dm:154]:
```suggestion
/obj/item/mecha_parts/mecha_equipment/weapon/energy/mecha_kineticgun/can_attach(obj/vehicle/sealed/mecha/M)
	. = ..()
	if(.) //combat mech
		return
	if(LAZYLEN(M.equipment) < M.max_equip)
		return TRUE
```
* Remember to preserve parent calls by default.
* No need of an `else` after an `if()` check that results on a `return`

**Comment by CloverIsLucky** [comment]
I have a question: Is the Exosuit KA moddable?

**Comment by InvalidArgument3** [comment]
> I have a question: Is the Exosuit KA moddable? I haven't tried it yet.

unfortunately not, and I don't have a clue how to implement it properly.

**Pull request merged**